datacube.py

```diff
@@ -0,0 +1,382 @@
+    - tuple: A tuple containing xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+    """
+    da_sen2: xr.DataArray = stackstac.stack(
+        items=s2_items[0],
+        epsg=epsg,
+        assets=S2_BANDS,
+        bounds_latlon=BBOX,  # W, S, E, N
+        resolution=resolution,
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    da_sen1: xr.DataArray = stackstac.stack(
+        items=s1_items[
+            1:
+        ],  # To only accept the same orbit state and date. Need better way to do this.
+        assets=["vh", "vv"],  # SAR polarizations
+        epsg=epsg,
+        bounds_latlon=BBOX,  # W, S, E, N
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    # To fix TypeError: Invalid value for attr 'spec'
+    da_sen1.attrs["spec"] = str(da_sen1.spec)
+
+    # To fix ValueError: unable to infer dtype on variable None
+    for key, val in da_sen1.coords.items():
+        if val.dtype == "object":
+            print("Deleting", key)
+            da_sen1 = da_sen1.drop_vars(names=key)
+
+    # Create xarray.Dataset datacube with VH and VV channels from SAR
+    # da_vh: xr.DataArray = da_sen1.sel(band="vh", drop=True).rename("vh")
+    # da_vv: xr.DataArray = da_sen1.sel(band="vv", drop=True).rename("vv")
+    # ds_sen1: xr.Dataset = xr.merge(objects=[da_vh, da_vv], join="override")
+    da_sen1 = stackstac.mosaic(da_sen1, dim="time")
+
+    # print(ds_sen1)
+
+    # da_sen1 = da_sen1.drop([da_sen1.time[0].values], dim='time') # Remove first scene which has ascending orbit
+
+    da_dem: xr.DataArray = stackstac.stack(
+        items=dem_items,
+        epsg=epsg,
+        bounds_latlon=BBOX,  # W, S, E, N
+        resolution=resolution,
+        xy_coords="center",  # pixel centroid coords instead of topleft corner
+        dtype=np.float32,
+        fill_value=np.nan,
+    )
+
+    da_dem = stackstac.mosaic(da_dem, dim="time")
+
+    # _, index = np.unique(da_dem['time'], return_index=True)  # Remove redundant time
+    # da_dem = da_dem.isel(time=index)
+
+    return da_sen2, da_sen1, da_dem
+
+
+def merge_datarrays(da_sen2, da_sen1, da_dem):
+    """
+    Merge xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+
+    Parameters:
+    - da_sen2 (xr.DataArray): xarray DataArray for Sentinel-2 data.
+    - da_sen1 (xr.DataArray): xarray DataArray for Sentinel-1 data.
+    - da_dem (xr.DataArray): xarray DataArray for DEM data.
+
+    Returns:
+    - xr.DataArray: Merged xarray DataArray.
+    """
+    da_merge = xr.merge([da_sen2, da_sen1, da_dem], compat="override")
+    print("Merged datarray: ", da_merge)
+    print(
+        "Time variables (S2, merged): ", da_sen2.time.values, da_merge.time.values
+    )  # da_sen1.time.values, da_dem.time.values
+    return da_merge
+
+
+def process(year1, year2, aoi, resolution, epsg):
```

Comment:
Not sure about needing to pass in an EPSG code here. For wide areas (in the West-to-East direction) that can span multiple UTM zones, wouldn't this force a single EPSG instead of multiple?

Reply:
We are trying to overlap data from S1 & DEMs over S2 tiles. As S2 tiles are following the MGRS format, each tile will fall under a single UTM zone (this is my understanding, correct me if I am wrong).
S1 & DEMs might span across multiple UTM zones, but stackstac should reproject all of them together.

Reply:
That makes sense to me. I guess I should switch this to just read the projection based on the MGRS tile's UTM zone instead of have a user argument for it.

Reply:
Reading the EPSG code directly from the Sentinel-2 item properties now.